### PR TITLE
add initrd-ssh for debugging boot issues

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

**Comment by Mic92** [comment]
Merging because comin keeps reverting this change otherwise.

File: modules/nixos/disko-zfs.nix [hunk 1]
```diff
@@ -13,6 +13,9 @@
     efiInstallAsRemovable = true;
   };
 
+  # Sometimes fails after the first try, with duplicate pool name errors
+  boot.initrd.systemd.services.zfs-import-zroot.serviceConfig.Restart = "on-failure";
```
Comment (zowoq) [inline modules/nixos/disko-zfs.nix:17]: @Mic92 Should we have this enabled by default?

File: modules/nixos/emergency-access.nix [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  # allow to access emergency shell with a password
+  boot.initrd.systemd.emergencyAccess = "$6$he2fblfl/H7I.kvz$WbSCMXu8ztmqfj5jG4czqvu/rkMHxufxqHgy1urzXFSN.jZB4QiW5lOjR08vk8pZTyim3TT1wFkMaNE9zZ3sc1";
```
Comment (zowoq) [inline modules/nixos/emergency-access.nix:3]: @Mic92 Should we have this enabled by default?

Comment (Mic92) [reply modules/nixos/emergency-access.nix:3]: Potentially. Back then, I was using `disko-install` to enable it.

Comment (Mic92) [reply modules/nixos/emergency-access.nix:3]: We can also enable ssh in the initrd as well so that we don't need to order a kvm switch every time.

**Pull request merged**